### PR TITLE
feat: add proposals api and view

### PR DESCRIPTION
**Pull request by baby230211**
## Types of changes
- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
- add proposals api url
- add proposals api view

## Steps to Test This Pull Request
1. Go to 'http://localhost:8000/api/proposals/'

## Expected behavior
response will be
1. num_proposed 
2. num_stage_1_reviews
3. num_stage_2_reviews

## Related Issue
#1127

### CODE REVIEW

**Comment by mattwang44** [comment]
- PR 不直接指到 master branch 嗎？
- flake8 沒過，記得修一下

File: src/proposals/api/views.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from reviews.models import Review
+
+class ProposalAPIView(views.APIView):
+
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        proposed_talk_num = TalkProposal.objects.count()
+        proposed_tutorial_num = TutorialProposal.objects.count()
```
Comment (mattwang44) [inline src/proposals/api/views.py:17]: 有個欄位 `cancelled` 代表 submitter 投稿後有沒有收回，應該要濾掉被收回的 proposal
ref: https://github.com/pycontw/pycon.tw/blob/master/src/proposals/models.py#L152-L156
```suggestion
        proposed_talk_num = TalkProposal.objects.filter(cancelled=False).count()
        proposed_tutorial_num = TutorialProposal.objects.filter(cancelled=False).count()
```

File: src/proposals/api/views.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        stage_1_reviews = Review.objects.filter(stage=1).count()
+        stage_2_reviews = Review.objects.filter(stage=2).count()
```
Comment (mattwang44) [inline src/proposals/api/views.py:19]: 建議統一 var name & response fields 為 `num_xxx`

File: src/proposals/api/views.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        num_proposal_talk = TalkProposal.objects.filter(cancelled=False).count()
+        num_proposed_tutorial = TutorialProposal.objects.filter(cancelled=False).count()
+        num_stage_1_reviews = Review.objects.filter(stage=1).count()
+        num_stage_2_reviews = Review.objects.filter(stage=2).count()
+
+        response_data = {}
+        response_data[num_proposal_talk] = num_proposal_talk
+        response_data[num_proposed_tutorial] = num_proposed_tutorial
+        response_data[num_stage_1_reviews] = num_stage_1_reviews
+        response_data[num_stage_2_reviews] = num_stage_2_reviews
```
Comment (mattwang44) [inline src/proposals/api/views.py:25]: 是不是該寫個 unit test XD

```python
num_proposal_talk = 1
response_data = {}
response_data[num_proposal_talk] = num_proposal_talk
print(response_data)  # {1: 1}
```

File: src/proposals/api/urls.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from django.urls import path
+from proposals.api.views import ProposalAPIView
+
+urlpatterns = [
+    path("", ProposalAPIView.as_view()),
```
Comment (mattwang44) [inline src/proposals/api/urls.py:5]: 讀者看到 `/api/proposals` 會覺得這隻 api 可能會直接給出 proposal data，覺得可以改成更好被解讀的 path，for example
```suggestion
    path("summary/", ProposalAPIView.as_view()),
```

File: src/proposals/api/views.py (hunk 3 above)
Comment (mattwang44) [reply src/proposals/api/views.py:25]: 建議直接改成一個 statement 就好
```suggestion
        response_data = {
            "num_proposal_talk": num_proposal_talk,
            ...
        }
```

Comment (baby230211) [reply src/proposals/api/views.py]: fixed ~

Comment (baby230211) [reply src/proposals/api/views.py]: added unit test

File: src/proposals/api/views.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+from rest_framework import views
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+
+from core.authentication import TokenAuthentication
+from proposals.models import TalkProposal, TutorialProposal
+from reviews.models import Review
+
+
+class ProposalAPIView(views.APIView):
```
Comment (mattwang44) [inline src/proposals/api/views.py:10]:
```suggestion
class ProposalSummaryAPIView(views.APIView):
```

File: src/proposals/api/views.py [hunk 6]
```diff
@@ -18,9 +18,10 @@ def get(self, request):
         num_stage_1_reviews = Review.objects.filter(stage=1).count()
         num_stage_2_reviews = Review.objects.filter(stage=2).count()
 
-        response_data = {}
-        response_data["num_proposed"] = num_proposed_talk + num_proposed_tutorial
-        response_data["num_stage_1_reviews"] = num_stage_1_reviews
-        response_data["num_stage_2_reviews"] = num_stage_2_reviews
+        response_data = {
+            "num_proposed": num_proposed_talk + num_proposed_tutorial,
```
[the remark this answers is not in the document]
Comment (baby230211) [reply src/proposals/api/views.py]: 沒，看來是我誤會了，我改回來

File: src/proposals/api/views.py (hunk 5 above)
Comment (baby230211) [reply src/proposals/api/views.py]: updated

**Review by mattwang44 — APPROVED** [review]
Nice work!

**Pull request merged**